### DISCUSSION TITLE
How can I convert color?

### DISCUSSION THREAD

**Discussion by qtbui159** [libvips/libvips#2793] [Q&A] [answered]
use ImageMagick:
convert a.png -fill transparent -opaque black b.png.

How can I use libvips to deal with that in c++?
thx

**Comment by jcupitt** [comment]
Hello @qtbui159,

Sorry, I've no idea what that command does. I tried it on an image here and it didn't seem to do anything.

Please explain what you are trying to achieve and give a sample input and output image.

**Comment by qtbui159** [comment]
> Hello @qtbui159,
> 
> Sorry, I've no idea what that command does. I tried it on an image here and it didn't seem to do anything.
> 
> Please explain what you are trying to achieve and give a sample input and output image.

Thanks for replying @jcupitt.
There is a simple thing. I just replace black(0,0,0,255) with transparent color(0,0,0,0). 

sample images:
[original png](https://github.com/qtbui159/IssueImage/blob/main/2022-05-09-1.png)
[expected png](https://github.com/qtbui159/IssueImage/blob/main/2022-05-09-2.png)

**Comment by jcupitt** [comment, marked as answer]
Ah OK. In python (for example), something like:

```python
import pyvips

image = pyvips.Image.new_from_file("a.png")
image = ((image == [0, 0, 0, 255]).bandand()).ifthenelse([0, 0, 0, 0], image)
image.write_to_file("b.png")
```

**Comment by qtbui159** [comment]
> Ah OK. In python (for example), something like:
> 
> ```python
> import pyvips
> 
> image = pyvips.Image.new_from_file("a.png")
> image = ((image == [0, 0, 0, 255]).bandand()).ifthenelse([0, 0, 0, 0], image)
> image.write_to_file("b.png")
> ```

Thx again. This code works very well !
